File: ruby/series/series.rb
class Series
  private

  attr_reader :digits

  def initialize(numbers)
    @digits = numbers.chars
  end

  def valid?(size)
    digits.length >= size
  end

  public

  def slices(size)
    raise SliceSizeError unless valid?(size)
    digits.each_cons(size).map(&:join)
  end
end

class SliceSizeError < ArgumentError
  def initialize
    raise self, 'Slice requested must be smaller than series.'
  end
end

File: python/series/series.py
def slices(series, length):
    validate(series, length)

    number_of_slices = len(series) - length + 1
    slices = []
    start = 0

    while start < number_of_slices:
        end = start + length
        slices.append(series[slice(start, end)])
        start += 1

    return slices

def validate(series, length):
    if len(series) <= 0 or length <= 0 or length > len(series):
        raise ValueError('.+')
